sw/device/silicon_creator/lib/drivers/alert.c

```diff
@@ -22,30 +22,38 @@ rom_error_t alert_configure(size_t index, alert_class_t cls,
   }
   index *= 4;
 
+  uint32_t reg_wr_count = 0;
+
   switch (cls) {
     case kAlertClassA:
       sec_mmio_write32_shadowed(
           kBase + ALERT_HANDLER_ALERT_CLASS_SHADOWED_0_REG_OFFSET + index,
           ALERT_HANDLER_ALERT_CLASS_SHADOWED_0_CLASS_A_0_VALUE_CLASSA);
+      ++reg_wr_count;
       break;
     case kAlertClassB:
       sec_mmio_write32_shadowed(
           kBase + ALERT_HANDLER_ALERT_CLASS_SHADOWED_0_REG_OFFSET + index,
           ALERT_HANDLER_ALERT_CLASS_SHADOWED_0_CLASS_A_0_VALUE_CLASSB);
+      ++reg_wr_count;
       break;
     case kAlertClassC:
       sec_mmio_write32_shadowed(
           kBase + ALERT_HANDLER_ALERT_CLASS_SHADOWED_0_REG_OFFSET + index,
           ALERT_HANDLER_ALERT_CLASS_SHADOWED_0_CLASS_A_0_VALUE_CLASSC);
+      ++reg_wr_count;
       break;
     case kAlertClassD:
       sec_mmio_write32_shadowed(
           kBase + ALERT_HANDLER_ALERT_CLASS_SHADOWED_0_REG_OFFSET + index,
           ALERT_HANDLER_ALERT_CLASS_SHADOWED_0_CLASS_A_0_VALUE_CLASSD);
+      ++reg_wr_count;
       break;
     case kAlertClassX:
+      sec_mmio_write_increment(reg_wr_count);
       return kErrorOk;
     default:
+      sec_mmio_write_increment(reg_wr_count);
```

Comment:
nit: do we need these calls? It looks like `reg_wr_count` will be zero (asking to make sure that I didn't miss something). I'm OK with keeping these to make sure that write count is updated on all return paths.

Reply:
I left them in for consistency, in case we want to move this into RETURN macros later on.